### ISSUE TITLE
[Change Requests] Redefine the role of GraphStorm's `DataLoaders`

### ISSUE THREAD

**Issue by zhjwy9343** [awslabs/graphstorm#756]
The current GraphStorm `DataLoader` relies on Dataset to provide task-specific data, e.g., subset of node features, labels. As the new `Dataset` will be task-agnostic, `DataLoader`s should contains such task-specific information, and query `Dataset` to obtain related data.

Sample code
```
train_dataloader = gs.dataloading.GSgnnNodeDataLoader(
    dataset=acm_data,
    target_idx=acm_data.get_node_train_set(ntypes=['paper']),
    node_feats={'author':['feat'], 'paper':['feat'],'subject':['feat']},
    label_field='label',
    fanout=[20, 20],
    batch_size=64,
    train_task=True)
val_dataloader = gs.dataloading.GSgnnNodeDataLoader(
    dataset=acm_data,
    target_idx=acm_data.get_node_val_set(ntypes=['paper']),
    node_feats={'author':['feat'], 'paper':['feat'],'subject':['feat']},
    label_field='label',
    fanout=[100, 100],
    batch_size=256,
    train_task=False)
```

**Request changes**

1. Add task specific arguments:
    1. `node_feats`, for node tasks `GSgnnNodeDataLoader`. The argument accepts 3 input formats:
        1. **a string**, for homogeneous graph with one feature field;
        2. **a list of strings**, for homogeneous graph with multiple feature fields.
        3. **a dictionary of list of strings**, for heterogeneous graph with any fields.
    2. `label_field`, for node tasks. The argument assumes only **ONE** node type is the target node type, and accepts 2 input format:
        1. **a string**, for one label field;
        2. **a list of strings**, for multiple label fields. 
    3. `edge_feats`, `label_field`, for `GSgnnEdgeDataLoader`. They accept the same input formats as `GSgnnNodeDataLoader`.
2.  Add property method `get_node_feats()` , returning a dictionary of strings(`ntypes`) and tensors(`features`).
3. Add property method  `get label_field()`, returning a dictionary of string(`target_ntype`) and tensors(`labels`).
